### ISSUE TITLE
CYAML_FIELD_FLOAT fails for `double *`

### ISSUE THREAD

**Issue by JamesDunne** [tlsa/libcyaml#67]
```
struct test1 {
    double *value;
};

static const cyaml_schema_field_t test1_fields_schema[] = {
	CYAML_FIELD_FLOAT(
		"value", CYAML_FLAG_OPTIONAL | CYAML_FLAG_POINTER,
		struct test1, value
	),
	CYAML_FIELD_END
};
```

When using `CYAML_FLAG_OPTIONAL | CYAML_FLAG_POINTER` flags, the data type becomes `double *` and not `double` and is hence *sized incorrectly*. This error exists for any usage of CYAML_FLAG_POINTER flag where the field type is a pointer to a primitive value and one of the FIELD_INT or FIELD_FLOAT macros is used.

The use case here is being able to preserve optionality of the value from YAML as a nullable pointer to a primitive value.

**Comment by tlsa** [comment]
Yes, you're quite right, thanks.  The `CYAML_FIELD_FLOAT` macro won't work for pointers.

I will add a `CYAML_FIELD_FLOAT_PTR` macro (and the same for the other types).

Note, you can still construct a correct schema directly, without using the `CYAML_FIELD_*` helper macros.  I was in two minds about including these helpers actually.  I'm not sure if they help or just hide what's going on.

**Comment by JamesDunne** [comment]
A `CYAML_FIELD_FLOAT_PTR` would work well assuming the `sizeof` was against a *dereferenced* member.

Yeah I see the macros are quite simple so I could just bypass it for this use-case and inline the initializers with appropriate fixes. Design trade-offs :)

**Comment by tlsa** [comment]
Yep, this distinction already existed for mappings (`CYAML_FIELD_MAPPING` and `CYAML_FIELD_MAPPING_PTR`), so I'm not quite sure why I didn't do it for everything. :smile_cat:

**Comment by tlsa** [comment]
Implemented in #69, thanks!
